master/internal/db/postgres_trial_profiler_metrics.go

```diff
@@ -55,3 +58,42 @@ OFFSET $2 LIMIT $3`, labelsJSON, offset, limit)
 	}
 	return pBatches, nil
 }
+
+const summaryWindowSeconds = 10 * 60
+
+// GetTrialProfilerMetricSummary gets a summary of profiler metrics.
+func (db *PgDB) GetTrialProfilerMetricSummary(
+	ctx context.Context, labels *trialv1.TrialProfilerMetricLabels,
+) (*trialv1.TrialProfilerMetricSummary, error) {
+	labelsJSON, err := protojson.Marshal(labels)
+	if err != nil {
+		return nil, errors.Wrap(err, "marshaling labels")
+	}
+
+	summary := trialv1.TrialProfilerMetricSummary{
+		Labels: labels,
+	}
+
+	if err := db.sql.QueryRowxContext(ctx, fmt.Sprintf(`
+WITH latest AS (
+	SELECT upper(m.ts_range) AS ts
+	FROM trial_profiler_metrics m
+	WHERE m.labels @> $1::jsonb
+	ORDER BY id DESC
+	LIMIT 1
+)
+SELECT
+  avg(q.v) AS avg,
+  -- classic signal-to-noise ratio
+  (avg(q.v) ^ 2.0) / (stddev(q.v) ^ 2.0) > 5.0 AS stable
```

Comment:
Why are we using signal-to-noise ratio here? It doesn't seem like a particularly appropriate measure to me. 

Signal-to-noise will return true on series that are obviously monotonically increasing (e.g. `[200, 210, 220, 230, 240, 250, 260, 270]`).

Reply:
To always return true on monotonically increasing series the measure would need a concept of time, which it doesn't, it just looks at a chunk of data without ordering.

Reply:
i didn't think the slope was very important, just the spread of the window. like, we wouldn't want to fire on everything with a clearly defined slope, unless it was big enough to cause a large spread in the distributed, which this would catch

Reply:
> this boolean is basically trying to detect whether the slope of the timeseries is flat or not

i think this is the root of disagreement, i just wanted/went with a measure of stability, not flatness